pages/en/lb4/Creating-components.md

```diff
@@ -307,3 +307,297 @@ class EmailComponent {
+[Server](Server.html).
+
+### A Basic Example
+The concept of servers in LoopBack doesn't actually require listening on ports;
+you can use this pattern to create workers that process data, write logs, or
+whatever you'd like.
+
+Let's create an example server! Please note that this is not a complete example:
+
+```ts
+import {Server} from '@loopback/core';
+import {processReports} from '../util';
+export class ReportProcessingServer implements Server {
+  @inject('database') db: DefaultCrudRepository<Report, int>;
+  interval: NodeJS.Timer;
+  async start() {
+    // Run the report processor every 2 minutes.
+    let window = Date.now();
+    interval = setInterval(async () => {
+      const results = await db.find({
+        createdDate: {
+          $gt: window,
+        },
+      });
+      if (results && results.length > 0) {
+        results.forEach(async (rep) => {
+          await db.updateById(rep.id, await processReport(rep));
+        }); 
+      }
+      window = Date.now();
+    }, 1000 * 120);
+    return Promise.resolve();
+  }
+
+  async stop() {
+    clearInterval(interval);
+    return Promise.resolve();
+  }
+}
+```
+In this example, the `ReportProcessingServer` will load new reports every two
+minutes from the database (using the provided repository), process those entries,
+and then update the database.
+
+### Using your server
+To use your server, simply bind it to your application's context.
+
+```ts
+import {ReportProcessingServer} from '../servers';
+
+export class MyApplication extends Application {
+  super();
+  // Will automatically start when app.start is called
+  this.server(ReportProcessingServer); 
+}
+```
+
+### A more advanced example
+Typically, you'll want server instances that listen for traffic on one or more
+ports (this is why they're called "servers", after all). This leads into a key
+concept to leverage for creating your custom servers.
+
+### Controllers and routing
+LoopBack 4 developers are strongly encouraged to use controllers for their
+modules, and this naturally leads to the concept of routing.
+
+No matter what protocol you intend to use for your custom server, you'll need
+to use some algorithm to determine _which_ controller and function to send
+request data to, and that means you need a router.
+
+For example, consider a "toy protocol" similar to the JSON RPC
+specification (but nowhere near as complete or robust).
+
+The toy protocol will require a JSON payload with three properties: `controller`,
+`method`, and `input`.
+
+An example request would look something like this:
+```json
+{
+  "controller": "GreetController",
+  "method": "basicHello",
+  "input": {
+    "name": "world",
+  }
+}
+```
+
+### Defining the controller
+Here is the `GreetController` and the `Person` type:
+
+{% include code-caption.html content="/src/controllers/greet.controller.ts" %}
+
+```ts
+export class GreetController {
+  basicHello(input: Person) {
+    return `Hello, ${input && input.name || 'World'}!`;
```

Comment:
It's a bad practice to mix business logic (what's the name to print) with views/templates. Please split this line into two:

```ts
const name = input && input.name || 'World';
return `Hello, ${name}!`;
```

It's not that bad here, but `hobbyHello` below is much more difficult to read in the current form. Here is what I am proposing for `hobbyHello`:

```ts
const hobby = input && input.hobby || 'underwater basket weaving';
return `${this.basicHello(input)} I heard you like ${hobby}.`;
```

(I think this will also make the code easier to debug, I am not sure if V8/DevTools Inspector can step through template parts.)